tests/python/test_bls.py

```diff
@@ -143,6 +143,33 @@ def copy(bls: ti.template(), w: ti.template()):
             assert w[i, j] == w2[i, j]
 
 
+@ti.require(ti.extension.bls)
+@ti.all_archs
+def test_bls_large_block():
+    n = 2**10
+    block_size = 2**5
+    stencil_length = 2**5
```

Comment:
Before this PR, `initialize_scratch_pad` in this test runs in $O(block_size^2 * stencil_length^2)$ time (which is 2**20, still super fast...).

After this PR, `initialize_scratch_pad` in this test runs in $O((block_size + stencil_length)^2)$ time (which is 2**12).

When I set either one of `block_size` or `stencil_length` to a value greater than `2**6`, I got cudaErrors. Is it the case that cuda doesn't support a BLS that is too large (i.e., > 2**12 f32s)?

Reply:
Yeah, but I think the BLS size is `(2x(2^6)) ^ 2 = 2^14`, since the length of each dimension is `block_size + stencil_length`. It seems that the standard shared memory limit is [48Kb per threadblock](https://www.google.com/search?q=cuda+shared+memory+max+size&rlz=1C5CHFA_enKR947KR947&oq=cuda+shared+memory+max+size&aqs=chrome..69i57.4356j0j7&sourceid=chrome&ie=UTF-8), so `2^14 * 4 = 64Kb` exceeds that.

Reply:
I see! I'm already using 16KB shared memory in this test. Maybe I should make it a little bit smaller to prevent using up all shared memory in the 16KB-per-block case?

Reply:
Right, I think it's fine so long as we don't exceed the 48Kb shared memory size.